synapse/_scripts/synapse_port_db.py

```diff
@@ -1133,6 +1138,27 @@ def r(txn: LoggingTransaction) -> None:
                 r,
             )
 
+    async def _setup_application_services_sequence(self) -> None:
+        curr_tnx_id: Optional[
+            int
+        ] = await self.sqlite_store.db_pool.simple_select_one_onecol(
+            table="application_services_txns",
+            keyvalues={},
+            retcol="COALESCE(max(txn_id), 0)",
+        )
+
+        def r(txn: LoggingTransaction) -> None:
+            txn.execute(
+                "ALTER SEQUENCE application_services_txn_id_seq RESTART WITH %s",
+                (curr_tnx_id + 1,),
+            )
+
+        if curr_tnx_id is not None:
+            await self.postgres_store.db_pool.runInteraction(
+                "_setup_application_services_sequence",
+                r,
+            )
```

Comment:
I think this will fix the mypy issue:

```suggestion
        if curr_tnx_id is not None:
            def r(txn: LoggingTransaction) -> None:
                txn.execute(
                    "ALTER SEQUENCE application_services_txn_id_seq RESTART WITH %s",
                    (curr_tnx_id + 1,),
               )

            await self.postgres_store.db_pool.runInteraction(
                "_setup_application_services_sequence",
                r,
            )
```